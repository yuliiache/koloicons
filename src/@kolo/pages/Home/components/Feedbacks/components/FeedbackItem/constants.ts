export const FEEDBACK_TEXT_TEST_ID = 'feedbackText';
