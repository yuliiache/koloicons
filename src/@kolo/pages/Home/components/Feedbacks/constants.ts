export const FEEDBACKS_KEY = 'feedbacks';
