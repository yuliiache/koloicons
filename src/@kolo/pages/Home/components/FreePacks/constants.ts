export const FREE_PACKS_KEY = 'freePacks';
export const FREE_PACKS_TEST_ID = 'packItemComponent';
