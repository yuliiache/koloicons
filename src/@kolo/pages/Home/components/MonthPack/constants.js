export const MONTH_PACK_KEY = 'monthPack';
