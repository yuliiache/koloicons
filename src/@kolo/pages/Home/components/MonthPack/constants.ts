export const MONTH_PACK_KEY = 'monthPack';
export const MONTH_PACK_TEST_ID = 'monthPackComponent';
