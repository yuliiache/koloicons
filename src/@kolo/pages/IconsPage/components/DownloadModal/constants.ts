export const ICON_KEY = 'icon';
