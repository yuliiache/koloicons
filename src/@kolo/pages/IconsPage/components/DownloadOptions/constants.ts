export enum IconSize {
  PX24 = '24px',
  PX30 = '30px',
  PX48 = '48px',
  PX60 = '60px',
  PX96 = '96px',
  PX120 = '120px',
  PX144 = '144px',
  PX150 = '150px',
  PX254 = '254px',
}
