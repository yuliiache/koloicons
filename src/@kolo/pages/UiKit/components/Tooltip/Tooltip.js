import React from 'react';
import PropTypes from 'prop-types';
import RcTooltip from 'rc-tooltip';

import {TOOLTIP_PLACEMENT, TRIGGER_ACTION} from './constants';
import {overlayInnerStyle} from './styles';

import 'rc-tooltip/assets/bootstrap_white.css';

export const Tooltip = ({label, placement = TOOLTIP_PLACEMENT.LEFT, trigger = TRIGGER_ACTION.HOVER, children}) => {
  if (!label) {
    return children;
  }

  return (
    <RcTooltip
      placement={placement}
      overlay={label}
      trigger={trigger}
      overlayInnerStyle={{...overlayInnerStyle}}
    >
      {children}
    </RcTooltip>
  );
};

Tooltip.propTypes = {
  label: PropTypes.string,
  placement: PropTypes.oneOf(Object.values(TOOLTIP_PLACEMENT)),
  trigger: PropTypes.oneOf(Object.values(TRIGGER_ACTION)),
  children: PropTypes.node,
};
