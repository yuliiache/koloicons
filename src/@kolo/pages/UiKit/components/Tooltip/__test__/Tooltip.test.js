import React from 'react';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';

import {TOOLTIP_PLACEMENT, TRIGGER_ACTION} from '../constants';
import {Tooltip} from '../Tooltip';

describe('Tooltip', () => {
  afterEach(cleanup);

  it('should render snapshot successfully with label', () => {
    Object.values(TOOLTIP_PLACEMENT).forEach((placement) => {
      const {getByText} = render(
        <Tooltip
          placement={placement}
          trigger={TRIGGER_ACTION.HOVER}
          label={placement}
        >
          {<button>{placement}</button>}
        </Tooltip>
      );

      const triggerElement = getByText(placement);

      fireEvent.mouseEnter(triggerElement);

      const tooltip = screen.queryAllByText(placement);

      expect(tooltip).toMatchSnapshot();
    });
  });

  it('should render snapshot successfully without label', () => {
    const {asFragment} = render(
      <Tooltip
        placement={TOOLTIP_PLACEMENT.LEFT}
        trigger={TRIGGER_ACTION.HOVER}
      >
        {<button>{TOOLTIP_PLACEMENT.LEFT}</button>}
      </Tooltip>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
