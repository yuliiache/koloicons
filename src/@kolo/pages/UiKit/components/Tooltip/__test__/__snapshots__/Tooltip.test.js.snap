// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render snapshot successfully with label 1`] = `
[
  <button>
    left
  </button>,
  <div
    class="rc-tooltip-inner"
    role="tooltip"
    style="padding: 8px 10px; color: rgb(255, 255, 255); text-align: left; text-decoration: none; background-color: rgb(55, 55, 55); border-radius: 6px; box-shadow: 0 0 4px rgba(0, 0, 0, 0.17); min-height: 34px;"
  >
    left
  </div>,
]
`;

exports[`Tooltip should render snapshot successfully with label 2`] = `
[
  <button>
    right
  </button>,
  <div
    class="rc-tooltip-inner"
    role="tooltip"
    style="padding: 8px 10px; color: rgb(255, 255, 255); text-align: left; text-decoration: none; background-color: rgb(55, 55, 55); border-radius: 6px; box-shadow: 0 0 4px rgba(0, 0, 0, 0.17); min-height: 34px;"
  >
    right
  </div>,
]
`;

exports[`Tooltip should render snapshot successfully with label 3`] = `
[
  <button>
    top
  </button>,
  <div
    class="rc-tooltip-inner"
    role="tooltip"
    style="padding: 8px 10px; color: rgb(255, 255, 255); text-align: left; text-decoration: none; background-color: rgb(55, 55, 55); border-radius: 6px; box-shadow: 0 0 4px rgba(0, 0, 0, 0.17); min-height: 34px;"
  >
    top
  </div>,
]
`;

exports[`Tooltip should render snapshot successfully with label 4`] = `
[
  <button>
    bottom
  </button>,
  <div
    class="rc-tooltip-inner"
    role="tooltip"
    style="padding: 8px 10px; color: rgb(255, 255, 255); text-align: left; text-decoration: none; background-color: rgb(55, 55, 55); border-radius: 6px; box-shadow: 0 0 4px rgba(0, 0, 0, 0.17); min-height: 34px;"
  >
    bottom
  </div>,
]
`;

exports[`Tooltip should render snapshot successfully without label 1`] = `
<DocumentFragment>
  <button>
    left
  </button>
</DocumentFragment>
`;
